flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericInMemoryCatalog.java

```diff
@@ -345,16 +345,18 @@ public void createFunction(ObjectPath functionPath, CatalogFunction function, bo
 		checkNotNull(functionPath);
 		checkNotNull(function);
 
-		if (!databaseExists(functionPath.getDatabaseName())) {
-			throw new DatabaseNotExistException(getName(), functionPath.getDatabaseName());
+		ObjectPath path = normalize(functionPath);
```

Comment:
should we do the normalizatoin in ObjectPath constructor (and any possible builders) so that we don't have to call normalize(functionPath) in many places?

It will also likely reduce the code diff.

Reply:
That's indeed a simpler way, but as mentioned in the description, I'm afraid that would impact other objects like tables. The current assumption in Flink is function names are case insensitive and other object names are case sensitive.

Reply:
okay. Another way to reduce the diff is using ObjectPath functionPath = normalize(functionPath), which only introduce a new line.